internal/instrument/instrument.go

```diff
@@ -965,23 +983,60 @@ func ginMiddleware(got *dst.AssignStmt) dst.Stmt {
 	if !ok {
 		return nil
 	}
+	stmt := useMiddleware(iden.Name, "GinMiddleware")
+	wrap(stmt)
+	return stmt
+}
+
+func isEcho(stmt *dst.AssignStmt) bool {
+	rhs := stmt.Rhs[0]
+	f, ok := funcIdent(rhs)
+	return ok && f.Path == "github.com/labstack/echo/v4" && f.Name == "New"
+}
+
+func echoMiddleware(got *dst.AssignStmt) dst.Stmt {
+	iden, ok := got.Lhs[0].(*dst.Ident)
+	if !ok {
+		return nil
+	}
+	stmt := useMiddleware(iden.Name, "EchoMiddleware")
+	wrap(stmt)
+	return stmt
+}
+
+func funcIdent(e dst.Expr) (*dst.Ident, bool) {
+	call, ok := e.(*dst.CallExpr)
+	if !ok {
+		return nil, false
+	}
+	f, ok := call.Fun.(*dst.Ident)
+	if !ok {
+		return nil, false
+	}
+	return f, true
+}
```

Comment:
Food for thought: as in my previous comment, the same goes for this kind of functions (also see `isUseMiddleware`) to navigate the AST. I feel they belong to their specific source code file.

Reply:
Agreed 👍

Reply:
In my next PR I leave these fucntions in `instrument.go`, as it seems a good place for them.